### ISSUE TITLE
Selective Downloading of Data

### ISSUE THREAD

**Issue by SHABINTHEGREAT** [charliewolf/pynder#64]
Is it possible to **download only a subset** or portion of the data?
For example, the data only for:
- **one matched user** : **`session.matches(i)`**
- **only the ping_time data** – for one user or all users : **`session.matches(i).user.ping_time`** or **`session.matches().ping_time`**
- **only the messages** – for one user or all users : **`session.matches(i).messages`** or **`session.matches().messages`**

instead of downloading the whole dataset of all users? : `session.matches()`

Given that the tinder app does not download all the data every time, and the tinder API does not seem to require downloading all the data, is it possible to download only a small subset?

Thank you!

**Comment by rdrapeau** [comment]
I also agree that this would be useful. I'm currently running into an issue where `session.matches()` is timing out with a 504 error because I have too many matches (would be my guess).

**Comment by charliewolf** [comment]
:+1: This is a good idea.

**Comment by SHABINTHEGREAT** [comment]
> _I'm currently running into an issue where session.matches() is timing out with a 504 error because I have **too many matches** (would be my guess)._

Just a question: **How many total matches** do you have that it begins to break down and throw errors?

Or could it be that the request times out because of slow internet speed or low bandwidth or due to any proxy servers in between which time out?

**Comment by rdrapeau** [comment]
I have 1300 total matches. Here is the output:

```
  matches = session.matches()

  File "/Library/Python/2.7/site-packages/pynder-0.0.10-py2.7.egg/pynder/session.py", line 32, in matches
    return [models.Match(m, self) for m in self._api.matches()]
  File "/Library/Python/2.7/site-packages/pynder-0.0.10-py2.7.egg/pynder/models/user.py", line 126, in __init__
    match['person']['_id'])['results']
  File "/Library/Python/2.7/site-packages/pynder-0.0.10-py2.7.egg/pynder/api.py", line 84, in user_info
    return self._get("/user/" + user_id)
  File "/Library/Python/2.7/site-packages/pynder-0.0.10-py2.7.egg/pynder/api.py", line 47, in _get
    return self._request("get", url)
  File "/Library/Python/2.7/site-packages/pynder-0.0.10-py2.7.egg/pynder/api.py", line 43, in _request
    raise errors.RequestError(result.status_code)
pynder.errors.RequestError: 504
```

> Or could it be that the request times out because of slow internet speed or low bandwidth or due to any proxy servers in between which time out?

I'd say that is the more likely case.

**Comment by charliewolf** [comment]
I think we can fix this by caching + lazyloading user data

**Comment by SHABINTHEGREAT** [comment]
The ability to download only a subset or portion of the data would make @rdrapeau's issue a non-issue. Plus, that will enable even more features and abilities for Pynder. Thoughts?

**Comment by SHABINTHEGREAT** [comment]
Dear Mr. Wolf,

Can you please implement a neat **`session.updates()`** just like the **`session.matches()`**
```
import datetime
lastActivity = datetime.datetime.utcnow().isoformat()
data = {'last_activity_date': lastActivity}
session._api._post('/updates', data)
```
This returns only the updates, apparently, which is a very small data footprint than calling session.matches() repeatedly.

_(And please do not be angry at me for trying to refer to the api this time! 😇 )_

**Comment by charliewolf** [comment]
I don't have time right now, but perhaps in the future, or if somebody else wants to open a PR they're more than welcome.

**Comment by SHABINTHEGREAT** [comment]
I really wish that I could, Mr. Wolf. But I do not know enough Python yet for that.
Perhaps, with a little help from you, I can try, test, debug, etc. _(Your expertise and my time! Plus, I can learn more Python too, which will be great!)_

**Comment by charliewolf** [comment]
See also #72 and #70

**Comment by h-2-0** [comment]
@SHABINTHEGREAT, @charliewolf 

This is a very good idea, Shabin. As a some sort of workaround to manipulate matches data I load **session._api.matches()** instead of loading **session.matches()**  because the latter _takes ages_ to load +200 matches whereas the former loads pretty fast. But the one problem I have found is that I  haven't figured it out how to send a message to a match using the data downloaded from session._api.matches().
